MyCryptoHQ/MyCrypto
# Custom DPaths, Improvements, and Fix SingularDTV

Closes #1311.

### Description / Changes

* Re-enables the ability to set a custom dpath.
* Shows the dpath of the hard-coded options in lighter text next the the label.
* Loosens up the dPath regex to allow for SingularDTV wallets. This is not BIP-44 compliant, but maintains functionality from the current prod site.

### Steps to Test

1. Run the app on `dpath-improvements` branch
2. Enter a mnemonic phrase, trezor, or ledger
3. Confirm that dpath dropdown shows dpaths as options
4. Select the "Custom" option from the dpath dropdown
5. Enter a custom dpath, confirm the addresses look correct (This is easily done by just punching in one of the hardcoded ones.)
6. Switch to SingularDTV, confirm that works.

### Screenshots

![screen shot 2018-03-20 at 12 56 25 pm](https://user-images.githubusercontent.com/649992/37672042-7b8b68d2-2c43-11e8-9b7c-08ebeb7537a8.png)

![screen shot 2018-03-20 at 1 27 06 pm](https://user-images.githubusercontent.com/649992/37672045-7d125db4-2c43-11e8-8663-c7bb67c9d5db.png)

![screen shot 2018-03-20 at 1 21 03 pm](https://user-images.githubusercontent.com/649992/37672049-7f45fe38-2c43-11e8-898d-4403d5da3ab4.png)

Just as a note, the reason there's a button and it doesn't just auto-refresh when a valid path is entered is because hardware wallets require you to confirm on-device. This would be annoying for typing it out manually, as your device would prompt you with every keystroke.
